api/src/pcapi/admin/custom_views/cine_office_pivot_view.py

```diff
@@ -71,11 +77,26 @@ def validate_form(self, form: Form) -> bool:
             id_at_provider=form.cinema_id.data, provider_id=cds_provider.id
         )
         if pivot and pivot.venueId != form.venue_id.data:
-            flash("Cet identifiant cinéma existe déjà pour un autre lieu")
+            flask.flash("Cet identifiant cinéma existe déjà pour un autre lieu")
             return False
 
         return super().validate_form(form)
 
+    def check_if_api_call_is_ok(self, account_id: str, api_token: str) -> None:
+        resource = "rating"
+        try:
+            api_url = f"https://{account_id}.{settings.CDS_API_URL}{resource}?api_token={api_token}"
+            response = requests.get(api_url)
+            if response.status_code == 200:
+                flask.flash("Connexion à l'API OK.")
+                return
+        except requests.exceptions.RequestException as exc:
+            logger.exception(
+                "Network error on checking CDS API information", extra={"exc": exc, "account_id": account_id}
+            )
```

Comment:
(détail) Est-ce qu'on ne pourrait pas laisser le code du connecteur construire l'URL (plutôt que de dupliquer le code ici), voire effectuer la requête ?

```suggestion
        try:
            cine_digital_service.get_resource(settings.CDS_API_URL, account_id, cinema_api_token, ResourceCDS.xxxx)
            flask.flash("Connexion à l'API OK.")
        except (requests.exceptions.RequestException, CineDigitalServiceAPIException) as exc:
            logger.exception(
                "Network error on checking CDS API information", extra={"exc": exc, "account_id": account_id}
            )
```

(Question au passage : a-t-on vraiment besoin de passer `settings.CDS_API_URL` à `get_resource()`, `put_resource()`, etc. Est-ce qu'on ne pourrait pas l'utiliser directement dans `_build_url() ?)

Reply:
Tout à fait je change cette partie en appelant le code du connecteur.

Pour la question sur l'argument `settings.CDS_API_URL`, on peut l'utiliser dans `_build_url()` c'est juste au début on pensait que la racine url sera différente pour chaque cinéma. Je me prévois un BSR pour changer ça.